ferry.go

```diff
@@ -288,6 +360,26 @@ func (f *Ferry) Initialize() (err error) {
 	f.DataIterator = f.NewDataIterator()
 	f.BatchWriter = f.NewBatchWriter()
 
+	if f.Config.VerifierType != "" {
+		if f.Verifier != nil {
+			return errors.New("VerifierType specified and Verifier is given. these are mutually exclusive options")
```

Comment:
Why do we support both? Wouldn't it be simpler to have a single codepath?

Reply:
I only have this because `Verifier` is an `interface`. While we don't do it in our use cases, you could conceivably configure and pass in your own verifier to the Ferry without having the Ferry initialize it for you. You could even create an entirely different verifier without having to change the Ferry.

The code after this doesn't really change with respect to how the verifier is initialized, so the only "complexity" should be the few lines of code here.